engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayProcessor.java

```diff
@@ -100,15 +100,19 @@ private Either<Failure, List<ExecutableSequenceFlow>> findSequenceFlowsToTake(
     }
 
     for (final ExecutableSequenceFlow sequenceFlow : element.getOutgoingWithCondition()) {
-      final Expression condition = sequenceFlow.getCondition();
-      final Either<Failure, Boolean> isFulfilledOrFailure =
-          expressionBehavior.evaluateBooleanExpression(condition, context.getElementInstanceKey());
-      if (isFulfilledOrFailure.isLeft()) {
-        return Either.left(isFulfilledOrFailure.getLeft());
-
-      } else if (isFulfilledOrFailure.get()) {
-        // the condition is fulfilled
-        executableSequenceFlows.add(sequenceFlow);
+      if (element.getDefaultFlow() != null && element.getDefaultFlow() == sequenceFlow) {
+        continue;
+      } else {
```

Comment:
🔧 This can be simplified while remaining logically equivalent 
```suggestion
      if (element.getDefaultFlow() == null || element.getDefaultFlow() != sequenceFlow) {
```